frontend/src/main/java/olumofe/raphael/frontend/CarController.java

```diff
@@ -0,0 +1,57 @@
+package olumofe.raphael.frontend;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.*;
+import starter.project.domain.Car;
+
+import java.util.*;
+
+@RestController
+@Component
+public class CarController {
+
+
+    private List<Car> carNewList = new ArrayList<>();
+
+    @GetMapping(path ="/car/{carId}", consumes = {"application/json"})
+    @ResponseBody
+    public Car getCars(@PathVariable("carId") int carId) {
+
+        for (Car car : carNewList) {
+            if (car.getId() == (carId)) {
+                Car returnedCar;
+                returnedCar = car;
+                return returnedCar;
+            }
+            }
+          return null;
+    }
+
+    @GetMapping(path = "/cars", consumes = {"application/json"})
+    @ResponseBody
+    private List<Car> getBatchCars(@RequestBody List<Integer> list) {
+        List<Car> newList = new ArrayList<>();
+    for(int i= 0; i <= list.size() -1; i ++){
+        if(carNewList.get(i).getId() == list.get(i)){
```

Comment:
This logic will only work if the cars in `carNewList` are defined in the same order as the list of the request body. Let me know if you have trouble spotting the bug but think about what the value `i` is in relation to the two lists.

Reply:
The methods been changed, thanks